### ISSUE TITLE
option to disable inner _messages queue

### ISSUE THREAD

**Issue by gwh-cpnet** [benbria/node-amqp-connection-manager#61]
I am well aware of that this repo is about maintaining the data flows. 

However in the edge environment, where the broker may not be accessible and robust as cloud environment, accumulating data at client side would consume all the memory and cause the system to break. 

I wish there is some strategies can be added to limit the memory usage.

**Comment by jwalton** [comment]
That's an interesting idea.  You could do this by supplying some function that evicts old messages from `_messages` when it gets too full?

**Comment by gwh-cpnet** [comment]
I have already made a work around in my project to pre-check _message content size, and shift it if oversize,  before publish new data into the channel.

```js
_.sumBy(channelWrapper._messages, 'content.length')
```

**Comment by zatziky** [comment]
We would need to completely disable the memory queuing and instead **fail fast**. We want our clients to stop sending requests as soon as rabbitMq is down. Do you plan to add this feature?

**Comment by jwalton** [comment]
If you have a PR that adds it in a clean way, I'd certainly merge it.

**Comment by zatziky** [comment]
You're right, I should add something to the community too. I will see if it's absolutely necessary. :D  @jwalton

**Comment by Pagebakers** [comment]
I'm running into issues with the internal queue too, since it has issues handling high throughput and is always running behind causing the memory to fill up untill the app crashes.
It would be a better idea only to use the internal _messages queue when the connections fails.

You can send messages directly by accessing _channel like this for example:

```js
      let sendToQueue
      if (channelWrapper._channel) {
        sendToQueue = channelWrapper._channel.sendToQueue
      } else {
        sendToQueue = channelWrapper.sendToQueue
      }
      
      sendToQueue(jobName(), new Buffer(EJSON.stringify(task)), options, (err) => {
        if (err) {
          channelWrapper.close()
          connection.close()
        }
      })
```

or without fallback just use this:
```js
      channelWrapper._channel.sendToQueue(jobName(), new Buffer(EJSON.stringify(task)), options, (err) => {
        if (err) {
          channelWrapper.close()
          connection.close()
        }
      })
```
